### ISSUE TITLE
docker image broken: Class path contains multiple SLF4J bindings.

### ISSUE THREAD

**Issue by anguslees** [mesos/elasticsearch#224]
docker image mesos/elasticsearch-scheduler:0.2.0 fails immediately with:

```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/tmp/elasticsearch-mesos-scheduler.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/tmp/elasticsearch-mesos-scheduler.jar!/lib/slf4j-log4j12-1.7.12.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:53)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Tomcat connector in failed state
	at org.springframework.boot.context.embedded.tomcat.TomcatEmbeddedServletContainer.start(TomcatEmbeddedServletContainer.java:157)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.startEmbeddedServletContainer(EmbeddedWebApplicationContext.java:288)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:141)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:483)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:118)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:686)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:320)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:139)
	at org.apache.mesos.elasticsearch.scheduler.Main.run(Main.java:69)
	at org.apache.mesos.elasticsearch.scheduler.Main.main(Main.java:47)
	... 6 more
```

**Comment by anguslees** [comment]
.. wait so does the previous 0.1.1 - what am I doing wrong?  (it's been many years since I last developed in Java..)

**Comment by philwinder** [comment]
Anguslees,
The error is not due to SLF4J, it's due to the Tomcat server not starting.
How are you starting the framework? Can you send your Marathon post? Or if you haven't submitted via marathon, can you you show us how you started?

Thanks, Phil

**Comment by anguslees** [comment]
Curious.  I'm starting it via this very boring marathon config:
```json
{
  "id": "elasticsearch-mesos-scheduler",
  "container": {
    "docker": {
      "image": "mesos/elasticsearch-scheduler:0.2.0",
      "network": "HOST"
    }
  },
  "args": [
    "-n", "3",
    "-zk", "zk://master.mesos:2181/mesos",
    "-ram", "2048"
  ],
  "cpus": 0.2,
  "mem": 512.0,
  "env": {
    "JAVA_OPTS": "-Xms128m -Xmx256m"
  },
  "instances": 1
}
```

I think I've fixed it purely by accident, however.  Some other docker-using jobs were failing to start following my upgrade to Mesos 0.23.0 / Marathon 0.10.0 - and I traced that down to mesos-slave now requiring `--docker=/usr/bin/docker` (previously the unqualified `--docker=docker` default worked just fine).  Fixing that (or perhaps the churn from rolling it out?) appears to have also made the above go away - but I confess I don't really understand why.

I'm going to close this as "unreproducible" for now - unless you want to morph it into a lower-priority bug to actually address the multiple SLF4J bindings.
